### ISSUE TITLE
Initial delay variable

### ISSUE THREAD

**Issue by modem7** [mattwebbio/orbital-sync#147]
### What feature would you like added?

Currently, if OS and Pihole start at the same time, due to the delay difference in waiting for Pihole to come up, a failure occurs:

`Failure: The host "http://<ip>/admin" refused to connect. Is it down?`

Requesting that a variable for delay be implemented so that on container start, OS will wait x period before doing the first operations.

**Comment by JonMills-MRWA** [comment]
Just make it dependent on the pihole container (& it being healthy)

    depends_on:
      pihole:
        condition: service_healthy

**Comment by acmor** [comment]
Thank you for your comment, but it did not work for me. On system boot I get sync failed e-mail.

On my pi3b I have 4 containers running:

- orbital-sync
- pihole
- watchtower
- protainer_agent

So I tried it with

>     depends_on:
>       pihole:
>         condition: service_healthy
>       watchtower:
>         condition: service_started
>       portainer_agent:
>         condition: service_started

but it didn't work for me neither...

**Comment by BuxtonCalvin** [comment]
Do you have a health check on your pi container?  Here's a simple check that will delay startup until there is container connectivity:
 ```
healthcheck:
    test: ["CMD", "dig", "+norecurse","+retry=0","@127.0.0.11","pi.hole"," || exit 1"]
    interval: 1m05s
    timeout: 10s
    retries: 0
    start_period: 40s```

**Comment by acmor** [comment]
> Do you have a health check on your pi container? Here's a simple check that will delay startup until there is container connectivity:

No, I haven't had that yet. Your HC seems to be working, thank you very much!

**Comment by BuxtonCalvin** [comment]
Make sure that "pi.hole" is your FQDN for your pihole container.  The dig command doesn't seem to work with "docker domain names" ie container names.  I use Cnames in pi-hole to this, and it looks something like pihole.local.foo.com  mapped to the FQDN of the server that the container stack sits on.  This should give you a healthy container status, which in turn will be picked up by your "depends on" "condition: service_healthy" clauses in your other containers.  Once the stack is created, you should be able to start and stop via portainer or the like without issue.

**Comment by acmor** [comment]
Just to be clear: the pihole container is healthy, I have no problem with that.
`pihole/pihole:latest             "/s6-init"               43 hours ago   Up About a minute (healthy)`

When deploying the DockerCompose it works wonderfully, the first sync runs without errors.
Only when I reboot do I get the error message and the first sync fails.
